### PR TITLE
feat(react-menu): fill icon on hover

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
Add styles to fill icon on hover:
Screenshot 2022-05-19 at 16 06 50

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                  Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `179.667 kB`<br />`50.151 kB` | `179.909 kB`<br />`50.245 kB` | `242 B` <br />`94 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `110.243 kB`<br />`33.543 kB` | `110.485 kB`<br />`33.633 kB` | `242 B` <br />`90 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `113.418 kB`<br />`34.011 kB` |  `113.66 kB`<br />`34.104 kB` | `242 B` <br />`93 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                  |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `34.078 kB`<br />`11.098 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/60804cec07a495b8a5d1fc66338bf0443b212144'>60804cec07a495b8a5d1fc66338bf0443b212144</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/Avatar-mount_base.html">1236</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/Avatar-mount.html">1313</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/Button-mount_base.html">755</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/Button-mount.html">759</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/FluentProvider-mount_base.html">2301</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/FluentProvider-mount.html">2322</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/FluentProviderWithTheme-mount_base.html">407</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/FluentProviderWithTheme-mount.html">402</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">329</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">429</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">419</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/MakeStyles-mount_base.html">1998</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23084/perf-test-react-components/MakeStyles-mount.html">1986</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-menu-6b954e4f-a2c3-40fc-ba75-5ddaf0e162b9.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Fill icon on hover",
```
Comment (ling1726) [inline change/@fluentui-react-menu-6b954e4f-a2c3-40fc-ba75-5ddaf0e162b9.json:3]:
```suggestion
  "comment": "feat: Fill icon on hover",
```

**Review by ling1726 — APPROVED** [review]
IMO since it's part of the official design, this should actually be in all stories that contain icons. However, I also accept that this is a terribly inefficient way of achieving intended design

**Comment by YuanboXue-Amber** [comment]
> IMO since it's part of the official design, this should actually be in all stories that contain icons. However, I also accept that this is a terribly inefficient way of achieving intended design

@ling1726 I updated all react-menu stories with icons. And yeah I think too this is not a scalable solution. For tmp we have bundled icon re-exported so it is not a problem. But I think icons should be exported as bundled from react-icons :/

**Pull request merged**